modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,291 @@
+ **/
+public class IncrementalDataPurging implements Runnable {
+    private static final Logger LOG = Logger.getLogger(IncrementalDataPurging.class);
+    private long purgeExecutionInterval = Expression.Time.minute(15).value();
+    private boolean purgingEnabled = true;
+    private Map<TimePeriod.Duration, Long> retentionPeriods = new EnumMap<>(TimePeriod.Duration.class);
+    private StreamEventPool streamEventPool;
+    private Map<TimePeriod.Duration, Table> aggregationTables;
+    private SiddhiAppContext siddhiAppContext;
+    private static ScheduledFuture scheduledPurgingTaskStatus;
+    private static final String INTERNAL_AGG_TIMESTAMP_FIELD = "AGG_TIMESTAMP";
+    private static final String EXTERNAL_AGG_TIMESTAMP_FIELD = "AGG_EVENT_TIMESTAMP";
+    private String purgingTimestampField;
+    private static final Long RETAIN_ALL = -1L;
+    private ComplexEventChunk<StateEvent> eventChunk = new ComplexEventChunk<>(true);
+    private List<VariableExpressionExecutor> variableExpressionExecutorList = new ArrayList<>();
+    private Attribute aggregatedTimestampAttribute;
+    private Map<TimePeriod.Duration, CompiledCondition> compiledConditionsHolder =
+            new EnumMap<>(TimePeriod.Duration.class);
+
+
+    public void init(AggregationDefinition aggregationDefinition, StreamEventPool streamEventPool,
+                     Map<TimePeriod.Duration, Table> aggregationTables, Boolean isProcessingOnExternalTime,
+                     SiddhiAppContext siddhiAppContext, Map<String, Table> tableMap) {
+        this.siddhiAppContext = siddhiAppContext;
+        List<Annotation> annotations = aggregationDefinition.getAnnotations();
+        this.streamEventPool = streamEventPool;
+        this.aggregationTables = aggregationTables;
+        if (isProcessingOnExternalTime) {
+            purgingTimestampField = EXTERNAL_AGG_TIMESTAMP_FIELD;
+        } else {
+            purgingTimestampField = INTERNAL_AGG_TIMESTAMP_FIELD;
+        }
+        aggregatedTimestampAttribute = new Attribute(purgingTimestampField, Attribute.Type.LONG);
+
+        VariableExpressionExecutor variableExpressionExecutor = new VariableExpressionExecutor(
+                aggregatedTimestampAttribute, 0, 1);
+        variableExpressionExecutorList.add(variableExpressionExecutor);
+        for (TimePeriod.Duration duration : aggregationTables.keySet()) {
+            switch (duration) {
+                case SECONDS:
+                    retentionPeriods.put(duration, Expression.Time.sec(30).value());
+                    break;
+                case MINUTES:
+                    retentionPeriods.put(duration, Expression.Time.hour(24).value());
+                    break;
+                case HOURS:
+                    retentionPeriods.put(duration, Expression.Time.day(30).value());
+                    break;
+                case DAYS:
+                    retentionPeriods.put(duration, Expression.Time.year(5).value());
```

Comment:
Why cannot we simply put the value 5 here?

Reply:
This approach was used because of the readability. otherwise, we have to put 5 years in milliseconds.